lib/puppet/provider/exec/powershell.rb

```diff
@@ -8,12 +8,10 @@
       "#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe"
     elsif File.exists?("#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe")
       "#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe"
-    elsif File.exists?('/usr/bin/powershell')
-      '/usr/bin/powershell'
-    elsif File.exists?('/usr/local/bin/powershell')
-      '/usr/local/bin/powershell'
-    elsif !Puppet::Util::Platform.windows?
```

Comment:
The default for non-windows platforms is needed because linux etc. won't have the file extension.

Reply:
You beat me to it...

Reply:
True. I think there was some question about when this location should ever be reached, but I guess if someone has deliberately installed to weird directory and put in `PATH` it would get here.